src/xdg-autostart-generator/xdg-autostart-service.c

```diff
@@ -444,8 +450,11 @@ int xdg_autostart_format_exec_start(
                 percent = strreplace(raw, "%", "%%");
                 if (!percent)
                         return log_oom();
+                tilde_expanded = strreplace(percent, "~", home);
```

Comment:
I do not think we should replace all `"~"`, e.g. `"foo~baz~aaa"`.
I think we should replace tilde only when the string starts with single tilde and remaining is empty or path:
- `"~"` -> `"/home/username"`
- `"~/path/to/file"` -> `"/home/username/path/to/file"`